File: archive-application/src/main/java/site/archive/dto/v2/MyArchiveListResponseDto.java
package site.archive.dto.v2;

import lombok.Getter;

import java.util.List;

@Getter
public class MyArchiveListResponseDto {

    private final Long archiveCount;
    private final List<MyArchiveResponseDto> archives;

    private MyArchiveListResponseDto(Long archiveCount, List<MyArchiveResponseDto> archives) {
        this.archiveCount = archiveCount;
        this.archives = archives;
    }

    public static MyArchiveListResponseDto from(Long archiveCount, List<MyArchiveResponseDto> archiveDtos) {
        return new MyArchiveListResponseDto(archiveCount, archiveDtos);
    }

}

File: archive-application/src/main/kotlin/site/archive/dto/v2/MyArchiveListResponseDto.kt
package site.archive.dto.v2

data class MyArchiveListResponseDto(
    val archiveCount: Long,
    val archives: List<MyArchiveResponseDto>
) {
    companion object {
        @JvmStatic
        fun from(archiveCount: Long, archiveDtos: List<MyArchiveResponseDto>): MyArchiveListResponseDto {
            return MyArchiveListResponseDto(archiveCount, archiveDtos)
        }
    }
}